force-app/main/default/classes/serviceReqTriggerHandler.cls

```diff
@@ -0,0 +1,67 @@
+/**  @classname             : ServiceReqTriggerHandler
+  *  @author                : Raghavendra Kadam
+  *  @date                  : 15-Mar-2023
+  *  @description           : This is Handler class of Trigger AssignServiceAgentOnRequest in which logic has written as 
+                              assignment service agent using queue respective to their Territories.
+  *  @modification log      : 
+  *  1. Raghavendra Kadam   :   Class and comments has written in this class 15-Mar-2023
+     2. Raghavendra Kadam   :   Code correction has been done as per review 17-Mar-2023
+*/
+public class ServiceReqTriggerHandler {
+
+/**  @methodname        : populateOwner
+  *  @author            : Raghavendra Kadam
+  *  @param             : List<Service_Request__c> as Trigger.New
+  *  @description       : This is Trigger Handler class method in which logic for assignment of owner is written
+  *  @return            : None
+*/
+  public static void populateOwner(List<Service_Request__c> newRequests)
+  {
+    List<String> contactIdList = new List<String>();
+    Map<ID, Contact> IdVsContactRecordMap = new Map<Id,Contact>();
+    
+    for(Service_Request__c sr : newRequests)
+    {
+      if(sr.Contact__c != null)
+      {
+          contactIdList.add(sr.Contact__c);
+      }
+    }
+
+    if(contactIdList.size() > 0 ){
+      IdVsContactRecordMap = new Map<ID, Contact>([SELECT Id, MailingCountry , MailingState, MailingPostalCode FROM Contact where Id IN : contactIdList]);
+    }
+
+
+    for(Service_Request__c sr : newRequests)
+    {
+      String currentRecordContCountry = IdVsContactRecordMap.get(sr.Contact__c).MailingCountry;
+      String currentRecordContState = IdVsContactRecordMap.get(sr.Contact__c).MailingState;
+      String currentRecordContZip = IdVsContactRecordMap.get(sr.Contact__c).MailingPostalCode;
+      String currentRecordServiceType = sr.Service__c;
+
+      Boolean isConStateZipPresent = currentRecordContCountry != '' && currentRecordContState != '' && currentRecordContZip != '';
+      Boolean isConStatePresent = currentRecordContCountry != '' && currentRecordContState != '' && currentRecordContZip == '';
+      Boolean isConPresent = currentRecordContCountry != '' && currentRecordContState == '' && currentRecordContZip == '';
+
+      List<Territory_Configuration__c> terConfigList = new List<Territory_Configuration__c>();
+
+      if(isConStateZipPresent || isConStatePresent || isConPresent){
+      terConfigList = [ SELECT Id, OwnerId, Name, Territory__c, Service__c 
```

Comment:
Please remove the query outside from for loop.

Reply:
- Done with the code change